server/src/test/java/org/cloudfoundry/identity/uaa/authentication/ZoneAwareWhitelistLogoutHandlerTests.java

```diff
@@ -163,12 +163,21 @@ public void test_external_client_redirect() {
     }
 
     @Test
-    public void test_exteral_logout() throws ServletException, IOException {
+    public void test_external_logout() throws ServletException, IOException {
         when(oAuthLogoutHandler.getLogoutUrl(null)).thenReturn("");
+        when(oAuthLogoutHandler.getPerformRpInitiatedLogout(null)).thenReturn(true);
         handler.onLogoutSuccess(request, response, null);
         verify(oAuthLogoutHandler, times(1)).onLogoutSuccess(request, response, null);
```

Comment:
I try to avoid using `verify`. Ideally we should be testing whether we are logged in the OIDC provider or not. Checking that we are calling a specific method is testing the implementation detail.

But, I think that this is one of the best possible examples of this pattern. Here, being logged out is pretty much equivalent to `oAuthLogoutHandler.onLogoutSuccess` being called. The parameters could change though...
Maybe better would be:

```
.onLogoutSuccess(any(), any(), any())
```

Reply:
The actual checking of "whether we are logged in the OIDC provider or not" is done in `OIDCLoginIT.java` which actually sets up the external OIDC integration. This test is just a lower-value but simpler unit test. 

"The parameters could change though...Maybe better would be: .onLogoutSuccess(any(), any(), any())"
Maybe? I'm trying to avoid non-essential refactors in this PR. The `onLogoutSuccess(request, response, null)` assertion is existing code, I don't wanna loosen the assertion in this PR.